### PR TITLE
feat: add API for renewal of the permission token

### PR DESCRIPTION
**Pull request by marcin-bazyl**
# COMPLETES #
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-475304

## This pull request addresses

There is no API to refresh permission token

## by making the following changes

Added API to refresh permission token.

NOTE: currently this doesn't work correctly for cases when we start an instant space meeting - the new meeting info we get after refresh is not the same as we got from /spaceInstant API call - this is being discussed with WbxAppAPI team

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -1420,6 +1395,92 @@ export default class Meeting extends StatelessWebexPlugin {
...
+      // @ts-ignore - config coming from registerPlugin
+      this.config.experimental.enableAdhocMeetings &&
+      // @ts-ignore
+      this.webex.meetings.preferredWebexSite;
+
+    const destination = isStartingSpaceInstantV2Meeting
+      ? this.meetingInfo.meetingJoinUrl
+      : this.destination;
+    const destinationType = isStartingSpaceInstantV2Meeting ? _MEETING_LINK_ : this.destinationType;
+
```
Comment (cscheffauer) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:1419]: thought: shall we add additional metrics here about that the `refresh` of the permission token has happened?

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 2]
```diff
@@ -1420,6 +1395,92 @@ export default class Meeting extends StatelessWebexPlugin {
...
+    await this.fetchMeetingInfoInternal({
```
Comment (cscheffauer) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:1420]: question: the part of catching the `captcha` and `password` errors is not in the internal fetchMeetingInfo function. We want to throw these errors when we refreshPermissionToken no? (for example the password changes while a user waits a long time in interstitial). It seems the captcha and password errors are still getting thrown in the call stack below, but then i am wondering why we had these handlers in line 1457 originally?

**Review by cscheffauer — CHANGES_REQUESTED** [review]
some questions before approving it

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 2 above)
Comment (marcin-bazyl) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:1420]: you mean the checks that we do in fetchMeetingInfo() before we call `this.attrs.meetingInfoProvider.fetchMeetingInfo()`? These are not possible to happen if we already have the permissionToken  (checked in line 1404). If we have the token, it means we must have passed correct password and captcha or they were not required. If password changes or for whatever reason wbxappapi decides we need to show a captcha when refreshing the token, the errors will be thrown by the call to `this.attrs.meetingInfoProvider.fetchMeetingInfo()` and are caught inside fetchMeetingInfoInternal() (line 1339)

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 1 above)
Comment (marcin-bazyl) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:1419]: yeah, good point, I forgot about the metrics, I thought there was a separate jira for that. I'll add them now.

Comment (marcin-bazyl) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:1424]: done

**Review by cscheffauer — APPROVED** [review]

**Pull request merged**